Source/Core/Common/Hash.cpp

```diff
@@ -530,4 +532,19 @@ void SetHash64Function()
     ptrHashFunction = &GetMurmurHash3;
   }
 }
+
+std::string MakeGameID(std::string file_name)
+{
+  const char* file_name_c = file_name.c_str();
```

Comment:
consider using std::string_view's `.size()` to determine if you're at the end of the string instead of relying on the string being null terminated.